### PR TITLE
add csr2dense and dense2csr backend to rocalution

### PR DESCRIPTION
**Pull request by jsandham**
Adding csr2dense and dense2csr back-end to rocalution

### CODE REVIEW

File: src/base/hip/hip_conversion.cpp [hunk 1]
```diff
@@ -637,6 +638,162 @@ namespace rocalution
...
+            allocate_hip(nrow * ncol, &temp);
+
+            rocsparse_status sparse_status = rocsparseTcsr2dense(
+            sparse_handle, nrow, ncol, src_descr, src.val, src.row_offset, src.col, temp, nrow);
+            CHECK_ROCSPARSE_ERROR(sparse_status, __FILE__, __LINE__);
+
+            ValueType alpha = static_cast<ValueType>(1);
+            ValueType beta = static_cast<ValueType>(0);
+
+            ValueType* B;
```
Comment (ntrost57) [inline src/base/hip/hip_conversion.cpp:678]: Doesn't B need to be allocated by the user?

File: src/base/hip/hip_conversion.cpp [hunk 2]
```diff
@@ -637,6 +638,162 @@ namespace rocalution
...
+        }
+        else
+        {
+            ValueType* temp = NULL;
+            allocate_hip(nrow * ncol, &temp);
+
+            ValueType alpha = static_cast<ValueType>(1);
+            ValueType beta = static_cast<ValueType>(0);
+
+            ValueType* B;
```
Comment (ntrost57) [inline src/base/hip/hip_conversion.cpp:753]: Doesn't B need to be allocated by the user?

File: src/base/hip/hip_conversion.cpp (hunk 1 above)
Comment (jsandham) [reply src/base/hip/hip_conversion.cpp:678]: I don't actually use B in the routine as I am just transposing the matrix (beta equals zero) but I cant pass NULL in for B in geam

Comment (ntrost57) [reply src/base/hip/hip_conversion.cpp:678]: Ah right, you are just transposing A

Comment (jsandham) [reply src/base/hip/hip_conversion.cpp:678]: i.e. the geam code detects that beta is zero and the corresponding kernel that is called does not use B

Comment (jsandham) [reply src/base/hip/hip_conversion.cpp:678]: I should add a comment probably explaining this though

Comment (jsandham) [reply src/base/hip/hip_conversion.cpp:679]: Added comment in latest push

File: src/base/hip/hip_conversion.cpp (hunk 2 above)
Comment (jsandham) [reply src/base/hip/hip_conversion.cpp:755]: See above

**Pull request closed without merging**